app/instrumentation-tests/src/org/commcare/androidTests/ConditionalEnumInCaseListTests.kt

```diff
@@ -0,0 +1,162 @@
+package org.commcare.androidTests
+
+
+
+import android.widget.DatePicker
+import androidx.test.espresso.Espresso.*
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.contrib.PickerActions
+
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+
+import org.commcare.annotations.BrowserstackTests
+import org.commcare.dalvik.R
+import org.commcare.utils.InstrumentationUtility
+
+import org.commcare.utils.isPresent
+import org.hamcrest.Matchers
+import org.junit.After
+
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.FixMethodOrder
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.MethodSorters
+import java.time.LocalDate
+import java.util.*
+
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+@BrowserstackTests
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)  //executes the tests in order
+class ConditionalEnumInCaseListTests: BaseTest() {
+    companion object {
+        const val CCZ_NAME = "conditional_enum_in_case_list.ccz"
+        const val APP_NAME = "Conditional Enum in Case List Tests"
+        val date = LocalDate.now().toString()
+        //the list contains the form input values of name, age, appointment date and appointment attended
+        val listOfCases = listOf(
+            listOf("Test One "+date, "16", "Today", "yes"),
+            listOf("Test Two "+date, "21", "Yesterday", "no")
+        )
+    }
+
+    @Before
+    fun setup() {
+        installApp(APP_NAME, CCZ_NAME)
+        InstrumentationUtility.login("test1", "123")
+    }
+
+    /**
+     * Adding the teardown method so the tests doesnot fail after only thefirst execution, when executed as a whole.
+     */
+    @After
+    fun tearDown(){
+        InstrumentationUtility.logout()
+        InstrumentationUtility.uninstallCurrentApp()
+    }
+    @Test
+    fun testCreateCases(){
+        for (caseList in listOfCases) {
+            InstrumentationUtility.openForm(0,0)
+            onView(ViewMatchers.withClassName(Matchers.endsWith("EditText")))
```

Comment:
can abstract as `InstrumentationUtility.enterText()`

Reply:
one utility was already present with same name that required element id, so created another without ID argument.